render-release-notes.sh

```diff
@@ -0,0 +1,174 @@
+        user's home directory with a generated CSS href relative path of
+        'styles/ReleaseNotes.css' for the link element in the head section of
+        the release notes HTML file. Finally, ReleaseNotes.css will be copied
+        to ~/styles.
+EOF
+	exit 1
+}
+
+function die() {
+	echo "$*" >&2
+	popd >/dev/null
+	exit 1
+}
+
+function invalid_arguments () {
+	die "
+Usage: $me --installer <inno|msi|nuget>
+   or: $me [--output <dir> [--css <path>]]
+Try '$me --help' for more information.
+"
+	exit 1
+}
+
+function render_release_notes () {
+	(homepage=https://git-for-windows.github.io/ &&
+		contribute=$homepage#contribute &&
+		wiki=https://github.com/git-for-windows/git/wiki &&
+		faq=$wiki/FAQ &&
+		mailinglist=mailto:git@vger.kernel.org &&
+		links="$(printf "%s\n%s\n%s\n%s\n%s\n%s\n%s\n%s\n%s\n" \
+		'<div class="links">' \
+		'<ul>' \
+		'<li><a href="'$homepage'">homepage</a></li>' \
+		'<li><a href="'$faq'">faq</a></li>' \
+		'<li><a href="'$contribute'">contribute</a></li>' \
+		'<li><a href="'$contribute'">bugs</a></li>' \
+		'<li><a href="'$mailinglist'">questions</a></li>' \
+		'</ul>' \
+		'</div>')" &&
+		printf '%s\n%s\n%s\n%s %s\n%s %s\n%s\n%s\n%s\n%s\n' \
+		'<!DOCTYPE html>' \
+		'<html>' \
+		'<head>' \
+		'<meta http-equiv="Content-Type" content="text/html;' \
+		'charset=UTF-8">' \
+		'<link rel="stylesheet"' \
+		' href="'$CSSPATH'">' \
+		'</head>' \
+		'<body class="details">' \
+		"$links" \
+		'<div class="content">'
+		markdown ReleaseNotes.md ||
+		die "Could not generate ReleaseNotes.html"
+		printf '</div>\n</body>\n</html>\n') >$TARGET
+}
+
+pushd $(pwd) >/dev/null
+
+for arg
+do
+	case "$1" in
+	-c|--css)
+		test -n "$INSTALLER" && invalid_arguments
+		shift
+		CSSPATH="${1%/}/"
+		shift
+		;;
+	-h|--help)
+		showUsage
+		;;
+	-i|--installer)
+		test -n "$OUTPUTDIR" && invalid_arguments
+
+		INSTALLER="$1"
+		shift
+		case "$1" in
+		inno)
+			OUTPUTDIR="installer/"
+			CSSPATH="usr/share/git/"
+			;;
+		msi)
+			OUTPUTDIR="msi/"
+			CSSPATH="usr/share/git/"
+			;;
+		nuget)
+			OUTPUTDIR="nuget/"
+			CSSPATH="content/"
+			;;
+		*)
+			invalid_arguments
+			exit 1
+			;;
+		esac
+		shift
+		;;
+	-o|--output)
+		test -n "$INSTALLER" && invalid_arguments
+
+		shift
+		OUTPUTDIR="${1%/}/"
+		shift
+		;;
+	esac
+done
+
+SCRIPT_PATH="$(cd "$(dirname "$0")" && pwd)"
+cd "$SCRIPT_PATH" || die "Could not switch directory to $SCRIPT_PATH"
+
+TARGET="${OUTPUTDIR-}ReleaseNotes.html"
+CSSPATH="${CSSPATH-}ReleaseNotes.css"
+
+# Generate the ReleaseNotes.html file
+test -x /usr/bin/markdown ||
+export PATH="$PATH:$(readlink -f "$PWD"/..)/../bin"
+
+# Install markdown
+type markdown ||
+pacman -Sy --noconfirm markdown ||
+die "Could not install markdown"
+
+test -z "$INSTALLER" && {
+	render_release_notes
+	test ! -f "$CSSPATH" && cp ReleaseNotes.css $CSSPATH
+} || {
+	test -f "$TARGET" &&
+	test "$TARGET" -nt ReleaseNotes.md &&
+	test "$TARGET" -nt "$OUTPUTDIR"/release.sh || {
```

Comment:
OH! I was wondering why you were "detecting changes" to `release.sh` -- I didn't know it was really about detecting changes to the HTML preamble. That makes sense. Thanks!!